### PR TITLE
Modals

### PR DESCRIPTION
**Pull request by alexcherman**

### CODE REVIEW

File: universal-login-wallet/src/ui/Modals/Modal.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  useEffect(() => {
+    const subscription = emitter.addListener('showModal', setModal);
+    return () => {
+      subscription.remove();
+    };
+  });
+
+  const listenKeyboard = (event : any) => {
+    if (event.key === 'Escape' || event.keyCode === 27) {
```
Comment (JustynaBroniszewska) [inline universal-login-wallet/src/ui/Modals/Modal.tsx]: `const KEY_CODE_ESCAPE = 27;`

Comment (marekkirejczyk) [reply universal-login-wallet/src/ui/Modals/Modal.tsx]: to commons

File: universal-login-wallet/src/ui/common/InputWithDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import useToggler from '../../hooks/useToggler';
+
+interface InputProps {
+  onChange: (...args: any[]) => void;
+  placeholder?: string;
+  autoFocus?: boolean;
+  className?: string;
+  id: string;
+}
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/ui/common/InputWithDropdown.tsx]: 👍

File: universal-login-wallet/src/utils/copy.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+function copy(id : string) {
+  const content = document.getElementById(id) as HTMLInputElement;
+  content.select();
+  document.execCommand('copy');
+}
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/utils/copy.ts]: commons

**Comment by alexvandesande** [comment]
ha, you thought you'd sneak this in? Did some digging and was able to check the working in progress. Looking nice so far!

File: universal-login-wallet/src/ui/common/InputLabel.tsx [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+
```
Comment (JustynaBroniszewska) [inline universal-login-wallet/src/ui/common/InputLabel.tsx]: `interface InputLabelProps {
   children: any;
   htmlFor: string;
}`

File: universal-login-wallet/src/ui/common/InputLabel.tsx [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+const InputLabel = ({children, htmlFor} : {children: any, htmlFor: string}) => (
```
Comment (JustynaBroniszewska) [inline universal-login-wallet/src/ui/common/InputLabel.tsx]: ```{chldren, htmlFor} : InputLabelProps```

File: universal-login-wallet/src/ui/common/InputText.tsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+interface InputTextProps {
+  onChange: (...args: any[]) => void;
+  placeholder?: string;
+  autoFocus?: boolean;
+  className?: string;
+  id: string;
+}
+
+const InputText = (props: InputTextProps) => (
```
Comment (JustynaBroniszewska) [inline universal-login-wallet/src/ui/common/InputText.tsx]: Replace ```props``` with ```{onChanfe, placeholder, autoFocus, className, id}```

File: universal-login-wallet/src/ui/common/InputWithDropdown.tsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+      <input
+        id={id}
+        className={`input input-dropdown ${className ? className : ''}`}
+        onChange={onChange}
+        type="number"
+        autoFocus={autoFocus}
+        placeholder={placeholder}
+      />
+      <div className="currency-dropdown">
+        <button onClick={toggle} className="currency-dropdown-btn">ETH</button>
```
Comment (JustynaBroniszewska) [inline universal-login-wallet/src/ui/common/InputWithDropdown.tsx]: <buttton ... > {shortcuts[0]} </button>

File: universal-login-wallet/src/ui/common/InputWithDropdown.tsx [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+        {visible ?
+          <ul className="currency-dropdown-list">
```
Comment (JustynaBroniszewska) [inline universal-login-wallet/src/ui/common/InputWithDropdown.tsx]: extract ```<ul> ... </ul>``` -> ```loadCurrencyList()```

**Pull request merged**